gvmtools/parser.py

```diff
@@ -333,10 +333,13 @@ def create_connection(
             cafile=cafile,
         )
 
-    return SSHConnection(
-        timeout=timeout,
-        hostname=hostname,
-        port=port,
-        username=ssh_username,
-        password=ssh_password,
-    )
+    ssh_connection_args = {
+        'timeout': timeout,
+        'hostname': hostname,
+        'username': ssh_username,
+        'password': ssh_password,
+    }
+    if port is not None:
```

Comment:
Can port ever be `None` in the parser? If not passed via argparse I really think it is always the default port and can't be None.

Reply:
Hmm.. I haven't looked at this. I just wanted to improve code reliability if this function is ever called with port as None. But this is a problem in another repository and if the changes are merged, this won't be a problem anymore. I can revert these changes if desired?

Reply:
Yes I really think it can never happen here therefore this change should be reverted.